adele-payment-manager/src/main/java/com/microservicesteam/adele/payment/PaymentManager.java

```diff
@@ -1,26 +1,39 @@
 package com.microservicesteam.adele.payment;
 
-import static com.microservicesteam.adele.payment.PaymentStatus.CREATED;
-
-import com.paypal.api.payments.Payer;
-import com.paypal.api.payments.RedirectUrls;
+import com.microservicesteam.adele.payment.paypal.PaymentRequestMapper;
+import com.microservicesteam.adele.payment.paypal.PaymentResponseMapper;
+import com.microservicesteam.adele.payment.paypal.PaypalProxy;
+import com.paypal.api.payments.Payment;
+import com.paypal.base.rest.PayPalRESTException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
+@AllArgsConstructor
 public class PaymentManager {
 
+    private final PaymentRequestMapper paymentRequestMapper;
+    private final PaymentResponseMapper paymentResponseMapper;
+    private final PaypalProxy paypalProxy;
+
     public PaymentResponse initiatePayment(PaymentRequest paymentRequest) {
 
-        Payer payer = new Payer();
-        payer.setPaymentMethod("paypal");
+        Payment payment = paymentRequestMapper.mapTo(paymentRequest);
+        try {
+
+            Payment createdPayment = paypalProxy.create(payment);
+            log.debug("Payment created successfully at PayPal");
+            return paymentResponseMapper.mapTo(createdPayment);
 
-        RedirectUrls redirectUrls = new RedirectUrls();
-        redirectUrls.setReturnUrl(paymentRequest.returnUrl());
-        redirectUrls.setCancelUrl(paymentRequest.cancelUrl());
+        } catch (PayPalRESTException e) {
+            log.error("Error at PayPal", e);
+            return PaymentResponse.failed();
+
+        } catch (Exception ex) {
```

Comment:
When this can happen? Or this is just to make sure, no exceptions propagate to the caller's level?

Reply:
yes